### PR TITLE
Staging

### PR DESCRIPTION
**Pull request by Kugelsicher**

### CODE REVIEW

File: UserApi/Startup.cs [hunk 1]
```diff
@@ -30,13 +30,14 @@ public void ConfigureServices(IServiceCollection services)
...
-                                builder =>
-                                {
-                                    builder.AllowAnyOrigin()
-                                        .AllowAnyHeader()
-                                        .AllowAnyMethod();
-                                });
+                options.AddPolicy( name: "_testingPolicy",
+                                builder => builder
+                                    .WithOrigins(
+                                        "http://localhost:4200"
```
Comment (nicholasvvong) [inline UserApi/Startup.cs]: Temporary cors until origin page is setup

**Review by nicholasvvong — APPROVED** [review]
Everything is ready for deployment, except the cors configuration.

File: Logic/Mapper.cs [hunk 2]
```diff
@@ -8,19 +8,27 @@ public static class Mapper
...
             repoUser.Username = user.Username;
             repoUser.FirstName = user.Firstname;
             repoUser.LastName = user.Lastname;
             repoUser.Email = user.Email;
-            repoUser.Permissions = user.Permissions;
+            try {
+                repoUser.DateOfBirth = DateTime.ParseExact(user.Dateofbirth, "yyyy-MM-dd", CultureInfo.InvariantCulture);
+            }
+            catch (FormatException) {
+                Console.WriteLine("Mapper.UserToRepoUser() failed due to an invalid date");
```
Comment (markmoorerev) [inline Logic/Mapper.cs:28]: Instead of writing to the console, you need a logger.

File: Logic/UserLogic.cs [hunk 3]
```diff
@@ -14,39 +17,22 @@ public UserLogic(RepoLogic repo)
...
-        /// </summary>
-        /// <param name="username"></param>
-        /// <returns>User</returns>
         public User GetUser(string useremail)
         {
-            var repoUser = _repo.GetUser(useremail);
+            var repoUser = _repo.GetUserByEmail(useremail);
             if(repoUser == null)
             {
                 Console.WriteLine("UserLogic.GetUser() was called with a nonexistant username.");
```
Comment (markmoorerev) [inline Logic/UserLogic.cs]: log this instead of using console.writeline().

File: Logic/UserLogic.cs [hunk 4]
```diff
@@ -71,53 +53,19 @@ public async Task<List<User>> GetUsers()
...
-        /// <param name="userid"></param>
-        /// <returns></returns>
-        public async Task<List<string>> GetFollowingMovies(Guid userid)
+        public async Task<List<string>> GetFollowingMovies(string userid)
         {
-            List<Repository.Models.FollowingMovie> repoFollowingMovies = await _repo.GetFollowingMovies(userid.ToString());
+            List<Repository.Models.FollowingMovie> repoFollowingMovies = await _repo.GetFollowingMovies(userid);
             if(repoFollowingMovies == null)
             {
                 Console.WriteLine("UserLogic.GetFollowingMovies() was called for a username that doesn't exist.");
```
Comment (markmoorerev) [inline Logic/UserLogic.cs]: log this instead of using console.writeline().

File: Logic/UserLogic.cs [hunk 5]
```diff
@@ -14,39 +17,22 @@ public UserLogic(RepoLogic repo)
...
             _repo = repo;
         }
-        /// <summary>
-        /// Adds a new User Object to repository.
-        /// Returns true if sucessful; false otherwise.
-        /// </summary>
-        /// <param name="user"></param>
-        /// <returns></returns>
+        
         public async Task<bool> CreateUser(User user)
```
Comment (markmoorerev) [inline Logic/UserLogic.cs:29]: What happened to the comments?

File: UserApi/Controllers/UserController.cs [hunk 6]
```diff
@@ -123,17 +140,14 @@ public async Task<ActionResult> DeleteUser(Guid userid, [FromBody] User user)
...
         /// <summary>
         /// Changes a user's permissionlevel up to an admin level's (3)
+        /// If updating permissions fails, return 404
+        /// If updating is sucessful, return 202
         /// </summary>
         /// <param name="userid"></param>
         /// <returns></returns>
         [HttpPost("addadmin/{userid}")]
-        public async Task<ActionResult> AddAsAdmin(Guid userid)
+        public async Task<ActionResult> AddAsAdmin(string userid)
```
Comment (markmoorerev) [inline UserApi/Controllers/UserController.cs:141]: You should still check that the userid isn't empty, null, etc.

**Review by markmoorerev — CHANGES_REQUESTED** [review]
lots of comments needed. plus you need to implement logging in those places where you are using console.writeline()

File: UserApi/Controllers/UserController.cs (hunk 6 above)
Comment (nicholasvvong) [reply UserApi/Controllers/UserController.cs:141]: The check is done in the repo layer, and will return false all the way back up to the controller, and return 404

File: Logic/UserLogic.cs (hunk 5 above)
Comment (nicholasvvong) [reply Logic/UserLogic.cs:29]: Comments moved to interface

**Review by markmoorerev — APPROVED** [review]

**Pull request merged**